### PR TITLE
Use new NSM Spiffe and Cert rotation library

### PR DESCRIPTION
**Pull request by avahahn**
Signed-off-by: Ava Hahn <a.hahn@f5.com>

# NOTE
This PR should be merged only after the following:
- NGINX Service Mesh 1.7 is released
- go.mod is updated to point to the release version of the relevant NSM libraries.

### Proposed changes
Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue here in this description (not in the title of the PR).

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: internal/k8s/controller.go [hunk 1]
```diff
@@ -645,10 +649,27 @@ func (lbc *LoadBalancerController) Run() {
...
-		err := lbc.spiffeCertFetcher.Start(lbc.ctx, lbc.addInternalRouteServer)
+		_, _, err := lbc.spiffeCertFetcher.Start(lbc.ctx)
+		lbc.addInternalRouteServer()
 		if err != nil {
 			glog.Fatal(err)
 		}
+
+		// wait for initial bundle
+		cert := <-lbc.spiffeCertFetcher.CertCh
+		lbc.syncSVIDRotation(cert)
```
Comment (sjberman) [inline internal/k8s/controller.go:660]: The previous code in spiffe.go that waited for this had a timeout involved so we didn't hang forever. We should do the same here.

File: internal/k8s/controller.go [hunk 2]
```diff
@@ -246,11 +247,14 @@ func NewLoadBalancerController(input NewLoadBalancerControllerInput) *LoadBalanc
 
 	lbc.syncQueue = newTaskQueue(lbc.sync)
 	if input.SpireAgentAddress != "" {
-		var err error
-		lbc.spiffeCertFetcher, err = NewSpiffeCertFetcher(lbc.syncSVIDRotation, input.SpireAgentAddress)
+		client, err := workloadapi.New(lbc.ctx, workloadapi.WithAddr("unix://"+input.SpireAgentAddress))
```
Comment (sjberman) [inline internal/k8s/controller.go:250]: What if our library just did this initialization in the constructor of the CertFetcher?

Comment (sjberman) [reply internal/k8s/controller.go:250]: Then we could likely remove NIC's dependency on the spiffe library altogether.

Comment (avahahn) [reply internal/k8s/controller.go:250]: Unfortunately the cert synchronization callback still requires workloadapi for the argument type. Would you like me to typedef that in our library? then we can remove it.

File: internal/k8s/controller.go [hunk 3]
```diff
@@ -245,11 +246,14 @@ func NewLoadBalancerController(input NewLoadBalancerControllerInput) *LoadBalanc
...
 	if input.SpireAgentAddress != "" {
-		var err error
-		lbc.spiffeCertFetcher, err = NewSpiffeCertFetcher(lbc.syncSVIDRotation, input.SpireAgentAddress)
+		lbc.spiffeCertFetcher, err = spiffe.NewX509CertFetcherFromAddr(
+			lbc.ctx,
+			input.SpireAgentAddress,
+		)
 		if err != nil {
-			glog.Fatalf("failed to create Spiffe Controller: %v", err)
+			glog.Fatalf("failed to initialize certfetcher: %v", err)
```
Comment (sjberman) [inline internal/k8s/controller.go:256]:
```suggestion
			glog.Fatalf("failed to initialize spiffe certfetcher: %v", err)
```

File: internal/k8s/controller.go [hunk 4]
```diff
@@ -645,10 +649,33 @@ func (lbc *LoadBalancerController) Run() {
...
 		}
+
+		// wait for initial bundle
+		timeoutch := make(chan bool, 1)
+		go func() { time.Sleep(time.Second * 30); timeoutch <- true }()
+		select {
+		case cert := <-lbc.spiffeCertFetcher.CertCh:
+			lbc.syncSVIDRotation(cert)
+		case <-timeoutch:
+			glog.Fatal("Failed to download initial trust bundle")
```
Comment (sjberman) [inline internal/k8s/controller.go:665]:
```suggestion
			glog.Fatal("Failed to download initial spiffe trust bundle")
```

File: internal/k8s/controller.go (hunk 2 above)
Comment (sjberman) [reply internal/k8s/controller.go]: No that's okay, I didn't realize it was used elsewhere.

**Review by sjberman — APPROVED** [review]

**Review by jasonwilliams14 — APPROVED** [review]

**Review by lucacome — APPROVED** [review]

**Pull request merged**